run.sh

```diff
@@ -340,68 +367,90 @@ function run_cloud_build(){
     topic_name="$function_name-pubsub-topic-logs-to-logzio"
 
     gcloud functions deploy $function_name --gen2 --region=$gcp_region  --retry --trigger-topic=$topic_name --entry-point=LogzioHandler --runtime=go121  --source=./cloud_function_go  --no-allow-unauthenticated --set-env-vars=token=$token --set-env-vars=type=$log_type --set-env-vars=listener=$listener_url
+    
+    echo "$cmd_create_cloud_build"
+
     if [[ $? -ne 0 ]]; then
         echo -e "[ERROR] [$(date +"%Y-%m-%d %H:%M:%S")] Failed to create Cloud Function."
         exit 1
+    else 
+        echo -e "[INFO] [$(date +"%Y-%m-%d %H:%M:%S")] Cloud Function: $function_name created."    
     fi
-	
-    echo "$cmd_create_cloud_build"
-    echo -e "[INFO] [$(date +"%Y-%m-%d %H:%M:%S")] Cloud Build Initialization is finished."
 }
 
 # Init script with proper message and display active user account
 
 function gcloud_init_confs(){
     user_active_account="$(gcloud auth list --filter=status:ACTIVE --format="value(account)")"
     echo -e "[INFO] [$(date +"%Y-%m-%d %H:%M:%S")] Your active account [${user_active_account}]"
-    echo -e "[INFO] [$(date +"%Y-%m-%d %H:%M:%S")] Choose Project ID"
-    _choose_and_set_project_id
-	echo -e "[INFO] [$(date +"%Y-%m-%d %H:%M:%S")] Project ID was updated."
 }
 
 
 # Choose and set project id
 # Error:
 #   Exit Code 1
-function _choose_and_set_project_id(){
+function choose_and_set_project_id(){
     array_projects=()
-    project_id=""
+    selected_projects=()
     count=0
-    for project in  $(gcloud projects list --format="value(projectId)")
+    echo -e "[INFO] [$(date +"%Y-%m-%d %H:%M:%S")] Choose and set project ids ..."
+    # List all projects and store them in an array
+    for project in $(gcloud projects list --format="value(projectId)")
     do
         count=$((count + 1))
         echo "[$count]:  $project"
         array_projects+=("$project")
     done
-    read -p "Please fill in number of project where you would like the integration to be deployed in: " mainmenuinput
-    count_projects=0
-    for value in "${array_projects[@]}"
-    do
-        count_projects=$((count_projects + 1))
-        if [ "$mainmenuinput" = "$count_projects" ]; then
-            project_id=$value
-            gcloud config set project $project_id
-            if [[ $? -ne 0 ]]; then
-                echo -e "[ERROR] [$(date +"%Y-%m-%d %H:%M:%S")] Failed to create Cloud Function."
-                exit 1
-            fi
-            echo -e "[INFO] [$(date +"%Y-%m-%d %H:%M:%S")] Integration will be launch in Project ID=$project_id"
-        fi
-    done
+    # Add the option to deploy to all projects
+    echo "[$((count + 1))]:  All Projects"
+
+    # Prompt the user to select multiple projects
+    read -p "Please fill in the numbers of the projects where you would like the integration to be deployed, separated by spaces (or type 'all' to select all projects): " -a mainmenuinput
```

Comment:
Not sure I understand the user experience.. when you write `Please fill in the numbers of the projects` you mean the project count? or the projects id's?